### ISSUE TITLE
connection.sendNotification for init/exit events

### ISSUE THREAD

**Issue by divyenduz** [graphql/graphiql#891]
First of all, thanks for all the awesome work in this repository. 

In `graphql-language-service-server` we can send `init`/`exit` and other notification events from server to client to easily build features like status bar in an editor like VSCode. 

Similarly to [`graphql-for-vscode`](https://github.com/kumarharsh/graphql-for-vscode/blob/master/src/client/extension.ts) where they have created a status bar as they have built a custom LSP server on top of [gql service](https://github.com/Mayank1791989/gql). 

Opinion on this or suggested alternatives? 

If we can lock on an API, I can take up the implementation part. 

Thanks!

**Comment by Neitsch** [comment]
Hey @divyenduz , I think the main problem of the LSP is that it has a custom implementation for the LSP protocol. Once that is fixed it should be a lot easier to make something like this happen. Does this answer your question? :)

**Comment by divyenduz** [comment]
Indeed, it does! Feel free to close this issue if you want to! Maybe re-open it as a feature request in the new home of the language server :)

**Comment by acao** [comment]
I plan on doing this, for example, for confirming that a users graphql-config has been found as well, or sending a more visible notification when it's not, and potentially even the related schema of the file they are editing?
